Development/sdp/sdp_grammar.cpp

```diff
@@ -44,6 +44,14 @@ namespace sdp
             return v.as_number(), web::json::visit(web::json::basic_ostream_visitor<char>(os), v), os.str();
         }
         inline web::json::value s2jn(const std::string& s) { auto v = web::json::value::parse(utility::s2us(s)); return v.as_number(), v; }
+        inline web::json::value
+        s2jn_no_leading_zero(const std::string& s)
+        {
+            // strip leading zeroes as they are not valid json and parse will throw,
+            // but they are allowed for sdp fields like session-id
+            auto str = regex_replace(s, std::regex("^(-?)(0+)(0|[1-9])", std::regex::extended), "$1$3");
```

Comment:
Is the extended POSIX grammar required for this? I would have thought the default (modified ECMAScript) grammar would be sufficient?